Fetch upstream

Epikest opened the pull request:
stop the transparency war by implementing both alpha channel transparency and weird hacky dark brown being ignored

Risewill23 committed:
Add files via upload (#323)

Commit:
temp transparency fixes (#319)

stop the transparency war by implementing both alpha channel transparency and weird hacky dark brown being ignored

thoroc committed:
debug logging on auth failed (#326)

Co-authored-by: Thomas Roche <thomas.roche@postcodelottery.co.uk>

LukachoLombardi committed:
Added Tor to feature list (#334)

thoroc committed:
debug logging for error on response (#340)

Co-authored-by: Thomas Roche <thomas.roche@postcodelottery.co.uk>

ixi2101 committed:
🔀 Update README.md (#342)

Stop directing people to use the cursed jpg transparency

thoroc committed:
more debug logs (#345)

Co-authored-by: Thomas Roche <thomas.roche@postcodelottery.co.uk>

Skyline-9 committed:
Updated Black Shield to For The Badge (#346)

glob-bruh committed:
Updated various grammer issues on README.md (#351)

Co-authored-by: Glob_Bruh <no-email@noemail.com>

Skyline-9 committed:
Replace b64 for black badge to svg (#352)

Risewill23 committed:
File Splitting for proxy and utils (#347)

* Uploading so I don't get too behind

* Uploading so I don't get too behind

* Black format and line fix

* Black format and line fix

* nox update

* nox update

* line fix

* Add files via upload

* GetRandomProxy to get_random_proxy

Risewill23 committed:
Fix Proxy Issues (#375)

* Fix proxy and Tor again

* Fix proxy

* Remove self. from some things

bjb28 committed:
Increase the use of log levels and include usernames. (#382)

derac committed:
fix poetry imports, missing websocket (#400)

Commit:
Transparency and config changes (#384)

* add legacy transparency option to example config

* start of changes to main.py

read config option for legacy transparency and begin changes on transparency handling

* update closest_color()

move support for both types of transparency out to closest_color in mappings.py

* small fix on name map

add reverse map for color 0 (darkest red) to stop "invalid color" messages

* nox

update comments for nox

* nox again

fix typo

* auto indent kinda sucks sometimes

* will nox ever be happy

can it even feel? are style changes even that big of a deal? is this even a description?

* change default from false to true

enable legacy_transparency by default so we dont get people complaining that the (69, 42, 0) method isnt working on a new version

thoroc committed:
Update CONTRIBUTING.md (#401)

pullgang committed:
Fix intense CPU usage issue (#398)

https://stackoverflow.com/a/58210476

KaiserBloo committed:
Fix some grammar and improve mac tutorial (#403)

- Changed instances of MacOS or MacOSX to ‘macOS’ as it is the correct capitalization

- Added Homebrew installation method for macOS (widely used mac package manager)

- A couple general readability improvements

JanDeDinoMan commented:
Hmm until now we supported transparency by using the alpha channel. I still this is the more clean solution instead of using a random color.

Commit:
Add required dependencies in order to build Pillow in Dockerfile (#419)

tyalie committed:
Fixing small typo in color mapping (#417)

light teal was previously assumed to be #00CC00, which is a bright
green, instead of the actuall #00CCC0

francerz committed:
Username on authorization success or fail (#409)

Huskydog9988 committed:
Allow user to specify tor proxy ip (#408)

maltejur committed:
fix websocket error (#428)

Risewill23 committed:
Personal Proxy Update (#383)

* Fix proxy and Tor again

* Fix proxy

* Remove self. from some things

* Personal Proxy Update

* Personal Proxy Update

* Personal Proxy Update

* Personal Proxy Update

* Personal Proxy Update

* Update main.py

* Line ending fix

* Fix missing self

* name = None

* Update proxy.py

* Update proxy.py

Commit:
change to proxy.py to fix failing to connect to websocket errors (#439)

* Update proxy.py

fix for "failed to connect to websocket" error

* nox

Randomblock1 committed:
Fix nox failure (#441)

* Fix nox failure

caused by recent PyPy update on GitHub Runner.

* whoops wrong branch

Randomblock1 committed:
Fix nox, really (#444)

* Fix nox failure

caused by recent PyPy update on GitHub Runner.

* whoops wrong branch

* Fix nox for real this time (?)

Until this gets merged into the main setup-nox action, this will do

Skyline-9 committed:
Add contributor list to README! (#447)

Skyline-9 committed:
Fixed redundancy (#448)
